src/ol/interaction/Draw.js

```diff
@@ -888,10 +887,46 @@ class Draw extends PointerInteraction {
     return sketchFeature;
   }
 
+  /**
+   * Append coordinates to the end of the geometry that is currently being drawn.
+   * This can be used when drawing LineStrings or Polygons. Coordinates will
+   * either be appended to the current LineString or the outer ring of the current
+   * Polygon.
+   * @param {!LineCoordType} coordinateExtension Linear coordinates to be appended into
+   * the coordinate array.
+   * @api
+   */
+  appendCoordinates(coordinateExtension) {
+    const mode = this.mode_;
+    let coordinates = [];
+    if (mode === Mode.LINE_STRING) {
+      coordinates = /** @type {LineCoordType} */ this.sketchCoords_;
+    } else if (mode === Mode.POLYGON) {
+      coordinates = this.sketchCoords_ && this.sketchCoords_.length ? /** @type {PolyCoordType} */ (this.sketchCoords_)[0] : [];
+    }
+
+    // Remove last coordinate from sketch drawing (this coordinate follows cursor position)
+    const ending = coordinates.pop();
+
+    // Append coordinate list
+    for (let i = 0; i < coordinateExtension.length; i++) {
+      this.addToDrawing_(coordinateExtension[i]);
+    }
+
+    // Duplicate last coordinate for sketch drawing
+    this.addToDrawing_(ending);
+    //this.addToDrawing_(coordinateExtension[coordinateExtension.length - 1]);
+  }
+
   /**
    * Extend an existing geometry by adding additional points. This only works
```

Comment:
Maybe this could be reworded a bit to be clearer, something like:
> Initiate draw mode by starting from an existing geometry which will receive new additional points. This only works...

Reply:
reworded